apps/trading/components/rewards-container/active-rewards.tsx

```diff
@@ -0,0 +1,530 @@
+import {
+  useActiveRewardsQuery,
+  useMarketForRewardsQuery,
+} from './__generated__/Rewards';
+import { useT } from '../../lib/use-t';
+import { addDecimalsFormatNumber, formatNumber } from '@vegaprotocol/utils';
+import classNames from 'classnames';
+import {
+  Icon,
+  type IconName,
+  Intent,
+  Tooltip,
+  VegaIcon,
+  VegaIconNames,
+  type VegaIconSize,
+  TradingInput,
+} from '@vegaprotocol/ui-toolkit';
+import { IconNames } from '@blueprintjs/icons';
+import {
+  DistributionStrategyDescriptionMapping,
+  DistributionStrategyMapping,
+  EntityScope,
+  EntityScopeMapping,
+  type Maybe,
+  type Transfer,
+  type TransferNode,
+  TransferStatus,
+  TransferStatusMapping,
+  DispatchMetric,
+  DispatchMetricDescription,
+  DispatchMetricLabels,
+  type RecurringTransfer,
+  EntityScopeLabelMapping,
+} from '@vegaprotocol/types';
+import { Card } from '../card/card';
+import { useMemo, useState } from 'react';
+import { useAssetDataProvider } from '@vegaprotocol/assets';
+
+export type Filter = {
+  searchTerm: string;
+};
+
+export const isActiveReward = (node: TransferNode, currentEpoch: number) => {
+  const { transfer } = node;
+  if (transfer.kind.__typename !== 'RecurringTransfer') {
+    return false;
+  }
+  const { dispatchStrategy } = transfer.kind;
+
+  if (!dispatchStrategy) {
+    return false;
+  }
+
+  if (transfer.kind.endEpoch && transfer.kind.endEpoch < currentEpoch) {
+    return false;
+  }
+
+  if (transfer.status !== TransferStatus.STATUS_PENDING) {
+    return false;
+  }
+
+  return true;
+};
+
+export const applyFilter = (transfer: Transfer, filter: Filter) => {
+  if (
+    transfer.kind.__typename !== 'RecurringTransfer' ||
+    !transfer.kind.dispatchStrategy?.dispatchMetric
+  ) {
+    return false;
+  }
+  if (
+    DispatchMetricLabels[transfer.kind.dispatchStrategy.dispatchMetric]
```

Comment:
(Non blocking): It sounds like @JonRay15 wanted to expand the filter logic. It might be worth using something like https://www.npmjs.com/package/fuse.js if it gets complex

Reply:
I will take a look. https://blog.logrocket.com/fuse-js-dynamic-search-react-app/ 
Once I enrich the transfers with market names and settlement asset the filtering logic would be easier